### DISCUSSION TITLE
Extend Harvested File Components from generated.wxs in my wxs file.

### DISCUSSION THREAD

**Discussion by goldcode** [wixtoolset/issues#7908] [Questions] [answered]
- I have to install a file under `Program Files\Company\Application` with Read/Write permissions for all. 
- This file is automatically harvested using HarvestDirectory and hence exists in an autogenerated fragment.
- i need to reference this file to also add the file permissions in my manually added fragment, like so:

```
<File Id="TestId" Name="Test.cfg" DiskId="1" Source="RepoPath\App\Test.cfg">
  <Permission User="Everyone" GenericRead="yes" GenericWrite="yes" />
</File>
```
- doing so, i would get a wix build error like duplicate component or file referenced in two places or similar. 

> so I guess the question is how can i extend the components of already harvested files to add other features(like permissions, firewall etc.)?

***
My current solution is to 
- harvest the directory.
- add a XSLT  transform to remove the file components to be extended from the generated.wxs file.
- manually add these file components in my.wxs.

But maybe there is a better way?

**Comment by robmen** [comment, marked as answer]
That's the way if you're using Heat. I go through similar in [Deployment Dojo episode 52 - https://www.youtube.com/live/XcUmwegYClg?si=zFqGy5NL729osMym](https://www.youtube.com/live/XcUmwegYClg?si=zFqGy5NL729osMym).

There is an easier option if you have the [HeatWave Build Tools](https://www.firegiant.com/docs/heatwave-build-tools/), and I cover that option in [Deployment Dojo episode 53 - Harvesting simplified by the HeatWave Build Tools for WiX v4](https://www.youtube.com/live/XcUmwegYClg?si=IZKeTpSZKSjjN6Ha)
